src/main/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinter.java

```diff
@@ -41,6 +41,20 @@
  */
 public final class AstTreeStringPrinter {
 
+    /**
+     * Enum to be used for test if comments should be printed.
+     */
+    public enum PrintOptions {
+        /**
+         * Comments has to be printed.
+         */
+        COMMENTS,
+        /**
+         * Comments has NOT be printed.
+         */
+        NONE
+    }
```

Comment:
> PrintOptions.WITHOUT_COMMENTS

Imagine we add a 3rd option, print WITH_CODE.
What does WITHOUT_COMMENTS mean alongside other options like WITH_COMMENTS and WITH_CODE? It is not very clear if WITHOUT_COMMENTS means it also enables or disables WITH_CODE.
This is why I chose the naming scheme NONE, to signify no other positive options are applied.

IMO, i think we should stay with this scheme.

> Comments has NOT be printed.

This wording looks directly related to old TRUE/FALSE values.
A better wording would be something along the lines `No extra print options activated`.